backend/app/ingestion/models.py

```diff
@@ -163,3 +164,23 @@ async def get_similar_n_chunks(
         )
 
     return results_dict
+
+
+async def rerank_chunks(
+    similar_chunks: dict[int, DocumentChunk],
+    query_text: str,
+    n_top_rerank: int,
+    request: Request,
+) -> dict[int, DocumentChunk]:
+    """
+    This function reranks the chunks using the cross-encoder
+    """
+    encoder = request.app.state.crossencoder
+    contents = similar_chunks.values()
+    scores = encoder.predict([(query_text, content.text) for content in contents])
+
+    sorted_by_score = [v for _, v in sorted(zip(scores, contents), reverse=True)][
+        :n_top_rerank
+    ]
+
+    return dict(enumerate(sorted_by_score))
```

Comment:
Currently this will return a list sorted along these lines: 

```
"0": {
      "file_name": "Electronic ticket receipt, October 08 for MR MARK BOTTERILL.pdf",
      "chunk_id": 2, ...
"1": {
      "file_name": "Electronic ticket receipt, October 08 for MR MARK BOTTERILL.pdf",
      "chunk_id": 2,```
Where 0 and 1 correspond to the scores?
What do you think about a) modifying the schema to include a `rank` field (in instances where cross-encoder is enabled) or b) simply to do an f-string manipulation so we return `f"Rank: {index}"`. The former feels a bit more robust. 

I only raise as the responses + fields might grow in complexity so being as explicit as possible about what the 0 / 1/ 2 number represents (the rank) might be useful.

Reply:
0, 1 are ranks. I like your idea of making it explicit though. And maybe even adding `cosine_distance` score and `rerank` score.

I'll make that fix

Reply:
Hmmm one schemas would be:

list({rank: int, document: DocumenChunk}) but that's clunky since if we want to filter by rank, we need to go through the whole list.

Other option is basically what we have but we add `rerank_score` to DocumentChunk schema.

I am starting to prefer the second one but let's discuss on Monday.

Reply:
Im going to merge this with option 2 in since it's just a feature branch and can make this change in a future PR if needed